internal/indexer/indexer.go

```diff
@@ -854,6 +866,497 @@ func (i *Indexer) indexReferenceToExternalDefinition(p *packages.Package, docume
 	return rangeID, true
 }
 
+type def struct {
```

Comment:
I had been thinking it might be nice to split up `def` into `interfaceDef` and `typeDef` (not good names) both for type safety (in terms of passing them to functions in the correct spot) but also we can probably reduce some casting if we do that.

Reply:
Paired on this, 2 separate types made a few call sites more type safe but bloated other parts than need to treat interfaces and types in the same way.